webpack.config.js

```diff
@@ -0,0 +1,245 @@
+let fs = require('fs');
+let path = require('path');
+let webpack = require('webpack');
+// let ExtractTextPlugin = require('extract-text-webpack-plugin');
+let CleanWebpackPlugin = require('clean-webpack-plugin');
+let ProgressBarPlugin = require('progress-bar-webpack-plugin');
+let MiniCssExtractPlugin = require('mini-css-extract-plugin');
+let HtmlWebpackPlugin = require('html-webpack-plugin');
+let PreloadWebpackPlugin = require('preload-webpack-plugin');
+let ReplacePlugin = require('webpack-plugin-replace');
+let CopyPlugin = require('copy-webpack-plugin');
+let WatchTimestampsPlugin = require('./config/watch-timestamps-plugin');
+let BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+
+let oldWrite = process.stderr.write;
+process.stderr.write = chunk => {
+	if (String(chunk).indexOf('DeprecationWarning:')!==-1) return;
+	return oldWrite.call(process.stderr, chunk);
+};
+
+function parseJson(text, fallback) {
+	try {
+		return JSON.parse(text);
+	}
+	catch (e) { }
+	return fallback || {};
+}
+
+function readFile(filename) {
+	try {
+		return fs.readFileSync(filename);
+	}
+	catch (e) {}
+}
+
+module.exports = function(_, env) {
+	let isProd = env.mode === 'production';
+	let nodeModules = path.join(__dirname, 'node_modules');
+	let componentStyleDirs = [
+		path.join(__dirname, 'src/components'),
+		path.join(__dirname, 'src/routes')
+	];
+
+	let babelRc = parseJson(readFile('.babelrc'));
+	babelRc.babelrc = false;
+	if (isProd) babelRc.presets[0][1].modules = false;
+
+	let manifest = parseJson(readFile('./src/manifest.json'));
+
+	return {
+		mode: env.mode || 'development',
+		entry: path.join(__dirname, 'config/client-boot.js'),
+		// entry: './src/index',
+		output: {
+			filename: isProd ? '[name].[chunkhash:5].js' : '[name].js',
+			chunkFilename: '[name].chunk.[chunkhash:5].js',
+			path: path.join(__dirname, 'build'),
+			publicPath: '/'
+		},
+		resolve: {
+			extensions: ['.ts', '.tsx', '.js', '.jsx', '.scss', '.css'],
+			alias: {
+				'app-entry-point': path.join(__dirname, 'src/index'),
+				style: path.join(__dirname, 'src/style'),
+				preact$: path.join(__dirname, 'node_modules/preact/dist/preact.js')
+			}
+		},
+		resolveLoader: {
+			alias: {
+				async: path.join(__dirname, 'config/async-component-loader')
```

Comment:
No, it is a component that wraps up dynamic imports. There are other ways to do this though if we want - my issue with dynamic import is that it's a Promise, which means it's impossible to perform a synchronous render even if the chunk is already available.

Reply:
I don’t understand. Let’s talk this through in the meeting tomorrow! This seems like a core thing.

Reply:
Yup, for sure. It's a bit opaque, and there is probably a way to at least use `require.ensure` here rather than the `async!` prefix, which reduces the amount of magic we have going on.

Another option is to dump manual async usage entirely and just let the existing mapping of `src/routes` handle it during build.